sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosAuthorizationTokenResolver.java

```diff
@@ -1,33 +1,33 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package com.azure.cosmos.implementation;
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.util.Beta;
 
 import java.util.Map;
 
 /**
  * This interface is for client side implementation, which can be used for initializing
- * AsyncDocumentClient without passing master key, resource token and permission feed.<br>
+ * CosmosAsyncClient without passing master key, resource token and permission feed.<br>
  * <br>
  * Each time the SDK create request for CosmosDB, authorization token is generated based on that
- * request at client side which enables creation of one AsyncDocumentClient per application shared across various users
+ * request at client side which enables creation of one CosmosAsyncClient per application shared across various users
  * with different resource permissions.
  */
-@FunctionalInterface
+@Beta(value = Beta.SinceVersion.V4_24_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
 public interface CosmosAuthorizationTokenResolver {
```

Comment:
who is the customer for this new public API?

as the future path of authentication for many customers is AAD why can't we direct the customer to use AAD?

Reply:
The customer that is requesting this API is using resource tokens currently and do not have an easy transition to use AAD at this time. They plan to do so in the future but for now they are stuck in V3 (which is pretty bad for lots of reasons) and to move to V4 they need this feature at parity with the previous SDK.
We can sync offline for more details.

Reply:
This still needs clear messaging that the feature/interface is only introduced for backwards-compatibility and customers should expect to move to AAD auth. I would prefer deprecated tag - but at least very clear communication in the comments.

Reply:
This feature is not about AAD auth vs anything else, it's about giving control to the user over what they want to pass in as part of the authorization token header. For instance rather than using our internal implementation to construct and add the AAD token, they can implement the logic themselves if they choose so.
In the particular use case we need to address we do need to facilitate resource tokens being interchanged but again, this is not just about that case. The TokenResolver class puts the user in charge as to what they want to pass in as part of the authorization header value.